### PR TITLE
Fix main thread assertion for tint color on text nodes, re-render tint-able nodes on hierarchy changes

### PR DESCRIPTION
**Pull request by rahul-malik**
- Avoid accessing `self.tintColor` off-main since this will trigger an
assertion and might be undefined behavior
- Trigger setNeedsDisplay in cases of hierarchy change and
tintColorDidChange in ASButtonNode, ASTextNode{1,2}
- Remove overwriting the tintColor of _ASTableViewCell. This actually breaks the typical inheritance behavior since setting it's tint color to the ASCellNode tintColor will always return a value which will be the `systemBlueColor` if it is unable to be determined

### CODE REVIEW

**Comment by rahul-malik** [comment]
@bolsinga - PTAL, this should be good to land now

**Comment by rahul-malik** [comment]
Will definitely update the title / commit message. I did verify the tests fail without these changes

**Comment by bolsinga** [comment]
> Will definitely update the title / commit message. I did verify the tests fail without these changes

Thanks for confirming!

File: Source/ASTableView.mm [hunk 1]
```diff
@@ -113,12 +115,14 @@ - (void)setElement:(ASCollectionElement *)element
     self.selectionStyle = node.selectionStyle; 
     self.focusStyle = node.focusStyle;
     self.accessoryType = node.accessoryType;
-    self.tintColor = node.tintColor;
```
Comment (bolsinga) [inline Source/ASTableView.mm]: Just for code historian sake, this is removing #764

**Pull request merged**